src/compiler/proposals-and-stabilization.md

```diff
@@ -0,0 +1,314 @@
+    the implementation is feasible and will not put undue burden on the maintainers of the compiler,
+    then they can second the MCP and implementation can proceed.
+  - This isn't necessary if the owner of the implementation is a member of the compiler team
+- Stabilising a language feature
+  - Follow the language team's process and have the stabiization PR reviewed by a member of the
+    compiler team
+- Reverting stabilisation of a language feature
+  - Follow the language team's process and have the revert PR reviewed by a member of the
+    compiler team
+
+### Targets
+
+- Proposing a new target
+  - **Propose using:** PR
+  - **Approve using:** r+ (compiler leads)
+  - Open a PR with the new target (w/ relevant documentation updates) and document adherence to the
+    [target tier policy][tier_policy] in the description. New targets must start as tier three
+  - New targets should be assigned to the compiler team co-leads to check for any licensing
+    concerns
+- Promoting a target
+  - **Propose using:** PR
+  - **Approve using:** r+ (compiler leads)
+  - Open a PR with the new target and document adherence to the [target tier policy][tier_policy]
+    in the description
+  - New targets should be assigned to the compiler team co-leads to ensure that any demands on
+    the project infrastructure are considered and checked with relevant teams
+- Demoting/removing a target
+  - **Propose using:** MCP
+  - **Approve using:** FCP
+  - Write an MCP describing why the target should be demoted/removed and once discussion has
+    concluded, an FCP can be started to approve the demotion/removal.
+- Changing target baseline (e.g. minimum Darwin or Windows version bump)
+  - **Propose using:** MCP
+  - **Approve using:** FCP
+  - Write an MCP describing why the target should have a change of baseline and once discussion has
+    concluded, an FCP can be started to approve the change of baseline.
+- Adding/removing target maintainers
+  - **Propose using:** PR
+  - **Approve using:** r+
+  - Open a PR with the changes to the target documentation and obtain an r+ from the reviewer.
+- Adding a target feature
+  - **Propose using:** PR
+  - **Approve using:** r+
+  - Open a PR adding the target feature and obtain an r+ from the reviewer.
+- Stabilising a target feature
+  - **Propose using:** PR
+  - **Approve using:** FCP
+  - Open a PR stabilizing the target feature and once the reviewer is happy with the changes,
+    an FCP can be started.
+
+### Lints, errors and warnings
+
+- Adding a new warning/error
+  - **Propose using:** PR
+  - **Approve using:** r+
+  - Open a PR with the implementation and obtain an r+ from the reviewer.
+- Adding a new lint group
+  - Follow the language team's process and have the implementation PR reviewed by a member of the
+    compiler team
+- Adding a new lint
+  - Follow the language team's process and have the implementation PR reviewed by a member of the
+    compiler team
```

Comment:
This makes the assumption that the lint is lang lint, but we lint that are purely compiler matter (e.g. [`explicit_builtin_cfgs_in_flags`](https://doc.rust-lang.org/rustc/lints/listing/deny-by-default.html#explicit-builtin-cfgs-in-flags)). We should probably differentiate both king of lints.

The same applies FCW or default lint level.

Reply:
That makes sense, I've taken a stab at making the distinction where a lint is under our purview and when it isn't